### PR TITLE
[WIP] [#175] update treatment of build status

### LINKED ISSUE #175: proposed: Simplify icon states and use microcopy for nuance
Regarding #174, I'm a bit concerned about the way the build state icons are accumulating, and proposing some follow-up edits.

icon-styles

--  

## CSS Tidy-up

There's small glitches around some of the icons, looks like a light gray background shade got applied behind the icons that should be removed (I'll take care of this).

--  

## Pending vs Running

Updating the `active` state to `running` to better match the statuses in Brigade's API makes sense, however I'm not sure about the visual treatment in the UI. 

icon-styles-2

* use of orange indicates a warning or a caution state (IMO)
* the clock icon isn't a clear distinction between waiting to start / running / finished

Essentially both the `active` and `running` states are saying to the user - **wait**, things are happening. I think the icon styles should be similar, and perhaps it's better to distinguish between the states in some text micro copy in the build metadata, rather than trying to convey these nuances in state via icon choices.

i.e. I propose something like this, to let text distinguish between what kind of 'waiting' is happening:

icon-styles-3

 @seanknox, what do you think?

--  

Notes:

* the pace of the spinners would be synced - that gif hurts my brain
* in addition to distinguishing between pending vs running, having text convey state would also improve the unknown (aka missing/expired logs) state, as that icon isn't clear enough on it's own

### PR DESCRIPTION
**Pull request by flynnduism**
Bunch of tidy ups for the list of build results:

- [x] use spinner icons for running + pending states
- [x] add text to elaborate on state (Running... / Initializing... etc)
- [x] remove line break glitch
- [x] small css tidy-ups 
- [x] more detailed, contextual metadata for builds

---  

Follow up to #174. Closes #175.

### CODE REVIEW

**Comment by flynnduism** [comment]
Thanks @ese!

I'm splitting the last item (sorting) into it's own issue ([#179](https://github.com/Azure/kashti/issues/179)) as it's holding up this PR and isn't crucial here.

**Comment by flynnduism** [comment]
Screenshot of the main changes:

screen shot 2018-04-03 at 4 42 25 pm

Builds now have additional text metadata, which details:

* what the status is (i.e. elaborating on the icon)
* if the build is complete or not
* how long the build took (if complete)
* how long ago the build started

# Timestamps:

I'm using [Moment](https://github.com/urish/angular2-moment) for _neater_ timestamps, but have also included the more detailed timestamp info, which is visible on hover like so:

screen shot 2018-04-03 at 4 57 55 pm

screen shot 2018-04-03 at 4 56 56 pm

# Unknown State

If Kashti doesn't know what the status of a build is, this is the fallback `unknown` state: 

screen shot 2018-04-03 at 4 59 54 pm


This would appear in the case of expired logs. @technosophos is this phrasing fair in terms of what's happening on the Brigade side?

File: src/app/project/project.component.html [hunk 1]
```diff
@@ -20,35 +20,48 @@ <h1>{{ project?.name }} </h1>
...
+          <span class="act-times" *ngIf="build.worker?.status; else unKnownState">
+
+            <!-- if ongoing -->
+            <span class="act-times-wrap" *ngIf="build.worker?.status == 'Pending'" title="Waiting for image pull to complete, and for container to start. {{ build.worker?.start_time | date:'long' }}">
+              <span class="act-time-duration">Initializing...</span>
+              <span class="act-time-ago right text-right">{{ build.worker?.start_time | amDifference: today :'minutes' : true | amDuration:'minutes' | lowercase }} ago</span>
             </span>
-            <span class="act-message ng-binding">
-              <em>#</em>{{ build.id }}
+            <span class="act-times-wrap" *ngIf="build.worker?.status == 'Running'" title="Building started {{ build.worker?.start_time | date:'long' }}">
```
Comment (bacongobbler) [inline src/app/project/project.component.html]: Would "Build started" be a more clear title?

**Comment by radu-matei** [comment]
Not sure if it is (or should be) part of this PR, but worth mentioning:

successfully-failed

Otherwise, everything looks great!

File: src/app/project/project.component.html (hunk 1 above)
Comment (flynnduism) [reply src/app/project/project.component.html]: Than "Running..."? Or than "Build started {{timestamp}}? 

Open to either.

**Comment by flynnduism** [comment]
@radu-matei I have opened #180 to check + address the yellow badge glitch you screenshotted above.

Merging this.

**Pull request merged**